### PR TITLE
OD12-540 US credit invoices subtract tax amount from grand total

### PR DESCRIPTION
**Pull request by astirpe**
For what I understand from the [official documentation](https://developers.taxjar.com/api/reference/?python#post-create-a-refund-transactio) the refunds transaction request requires negative amounts.

@StefanRijnhart this PR is opened onto your branch https://github.com/VanMoof/account-fiscal-rule/tree/add/12.0/taxjar_salestax

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
You are right, taxjar expects a negative amount for refund invoices. That inversion is carried out here already: https://github.com/VanMoof/account-fiscal-rule/blob/add/12.0/taxjar_salestax/taxjar_salestax/models/account_invoice_line.py#L37

But it looks like the sign is not applied again when the tax amount comes back. In Odoo 8, we applied the sign again when storing the tax amount on the invoice lines: https://github.com/VanMoof/odoo-addons/blob/8.0/taxjar_salestax/models/account_invoice.py#L203

In Odoo 12, that inversion seems to be missing here: https://github.com/VanMoof/account-fiscal-rule/blob/add/12.0/taxjar_salestax/taxjar_salestax/models/account_invoice.py#L181

It looks like the amounts you are updating, in the create_order call (which is used on confirming an invoice rather than fetching the tax amounts for orders or invoices) did not have the inversion applied indeed. So while I think that your PR fixes something that is wrong (and was equally wrong in Odoo 8), I think the invoice line tax amount needs fixing as well.

File: taxjar_salestax/models/account_invoice.py [hunk 1]
```diff
@@ -236,14 +236,18 @@ def _taxjar_commit(self):
         lines = self.invoice_line_ids._get_taxjar_lines(
             config=config, commit=True)
 
+        amount_untaxed = self.amount_untaxed
+        taxjar_amount = self.taxjar_amount
         if self.type == 'out_refund' and self.refund_invoice_id:
             refund_reference = self.refund_invoice_id.number
+            amount_untaxed = -1 * amount_untaxed
+            taxjar_amount = -1 * taxjar_amount
```
Comment (StefanRijnhart) [inline taxjar_salestax/models/account_invoice.py:243]: Also apply the inversion if there is no refund_invoice_id on the invoice.

**Comment by StefanRijnhart** [comment]
Contradicting my own point, I see there is an `abs` applied to the invoice line tax amounts: https://github.com/VanMoof/account-fiscal-rule/blame/add/12.0/taxjar_salestax/taxjar_salestax/models/account_invoice.py#L175. This seems to be my code, but for sure applying `abs` is a bad practice in financial code! But now I'm not sure what to do in this case.

You say you haven't tested your code. Is it an option for you to do so and verify that your fix also fixes the total invoice amount in Odoo?

File: taxjar_salestax/models/account_invoice.py [hunk 2]
```diff
@@ -178,7 +178,7 @@ def _call_taxjar(self):
 
         for line_id, amount in items:
             line = self.env['account.invoice.line'].browse(line_id)
-            line.taxjar_amount = amount
+            line.taxjar_amount = sign * amount
```
Comment (astirpe) [inline taxjar_salestax/models/account_invoice.py:181]: @StefanRijnhart I also changed this line

**Comment by MPishpecki** [comment]
Ill deploy this to test. If it fails the review no biggie - we remove it.

**Comment by StefanRijnhart** [comment]
@MPishpecki it depends on the confidence that @astirpe has in the fix. Andrea, maybe you can fetch the sandbox settings from the test environment and do some debugging with a live connection to Taxjar's sandbox? That's what I did when developing and migrating the module.

**Comment by astirpe** [comment]
This PR is safe enough to be merged IMO, too bad that it was not deployed on the test server. The finance team was eager to run a smoke test yesterday.

**Comment by MPishpecki** [comment]
I will deploy on test and see how the test goes.

**Comment by astirpe** [comment]
For what I understood, manual tests passed and this PR is going to be deployed on production. So I think we should consider to merge this PR into branch "`/add/12.0/taxjar_salestax`"

**Review by MPishpecki — APPROVED** [review]

**Pull request merged**